flocker/testtools/__init__.py

```diff
@@ -299,6 +302,28 @@ def loop(failure):
     return d
 
 
+def retry_effect_with_timeout(effect, timeout):
+    """
+    If ``effect`` fails, retry it until ``timeout`` expires.
+
+    :param Effect effect: The Effect to retry.
+    :param int timeout: Timeout in seconds.
+    """
+
+    end_time = time.time() + timeout
+
+    def ret(e):
+        if time.time() > end_time:
+            return Effect(Constant(False))
+        else:
+            time.sleep(ret.wait_secs)
+            ret.wait_secs *= 2
+            return Effect(Constant(True))
+    ret.wait_secs = 1
```

Comment:
Please make the initial retry period a documented parameter. It's OK to make it default to 1.